pkg/app/web/src/components/deployments-page/deployment-filter/index.tsx

```diff
@@ -223,6 +226,41 @@ export const DeploymentFilter: FC<DeploymentFilterProps> = memo(
             ))}
           </Select>
         </FormControl>
+
+        <FormControl className={classes.formItem} variant="outlined">
+          <Autocomplete
+            multiple
+            autoHighlight
+            id="labels"
+            noOptionsText="Invalid label"
+            options={labelOptions}
+            value={selectedLabels}
+            onInputChange={(_, value) => {
+              const label = value.split(":");
+              if (label.length !== 2) return;
+              if (label[0].length === 0) return;
+              if (label[1].length === 0) return;
+              setLabelOptions([value]);
+            }}
+            onChange={(_, newValue) => {
+              setLabelOptions([]);
+              setSelectedLabels(newValue);
+              handleUpdateFilterValue({
+                labels: newValue,
+              });
+            }}
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                variant="outlined"
+                label="Labels"
+                margin="dense"
+                placeholder="key:value"
+                fullWidth
+              />
+            )}
+          />
+        </FormControl>
```

Comment:
I'm not sure, just want to confirm this label's value.
Are the labels which were specified in the query URL being shown up?

For example, when we enter this URL on the browser, we expect `key:value` to be displayed.

`https://domain.dev/deployments?labels%5B0%5D=key%3Avalue&page=1&project=test`

Reply:
Yes, it can be like `http://domain.dev/deployments?labels%5B0%5D=env%3Adev&project=test` as the same as the applications page.

The format is specified at https://github.com/pipe-cd/pipe/pull/3000/files#diff-259a276962e94e5c49e6edf8b83bb34d9aa13337cf0c7c591e2bd96c835014e0R128

Reply:
I think your given [link](https://github.com/pipe-cd/pipe/pull/3000/files#diff-259a276962e94e5c49e6edf8b83bb34d9aa13337cf0c7c591e2bd96c835014e0R128) is for embedding the labels in the URL, isn't it?
When the user sets a new label in the Filter the URL will be updated to reflect that label.

But, what I was talking about is the opposite thing. 
Will the Filter show the corresponding label when the user enters a URL containing a label?

Reply:
Good point! Currently, the label specified in the URL doesn't appear on the filtering form, even though others are shown.
This is not just for the deployment filter, but for the application one, hence let me address it as another PR.

Reply:
Opened https://github.com/pipe-cd/pipe/issues/3001

Reply:
Sure. Let merge this PR. Nice work!